lib/init/grass.py

```diff
@@ -1307,17 +1268,10 @@ def load_gisrc(gisrc, gisrcrc):
     return mapset_settings
 
 
-def can_start_in_gisrc_mapset(gisrc, ignore_lock=False):
+def can_start_in_gisrc_mapset(mapset_settings, ignore_lock=False):
     """Check if a mapset from a gisrc file is usable for a new session"""
     from grass.grassdb.checks import can_start_in_mapset
 
-    mapset_settings = MapsetSettings()
-    kv = read_gisrc(gisrc)
-    mapset_settings.gisdbase = kv.get("GISDBASE")
-    mapset_settings.location = kv.get("LOCATION_NAME")
-    mapset_settings.mapset = kv.get("MAPSET")
-    if not mapset_settings.is_valid():
-        return False
     return can_start_in_mapset(
         mapset_path=mapset_settings.full_mapset, ignore_lock=ignore_lock
     )
```

Comment:
What is the purpose of `can_start_in_gisrc_mapset` since it doesn't use gisrc after your change?

Reply:
None. I removed this function and used directly `can_start_in_mapset`.